commit fe3f7e6c2aa744f96992aa6d2aa991b30e6d02bf
Author: Yang Bo <yangbo@linuxdeepin.com>
Date:   Sun Jan 3 19:33:24 2016 +0800

    UBUNTU: SAUCE: (no-up) hv: Supply vendor ID and package ABI
    
    BugLink: http://bugs.launchpad.net/bugs/1193172
    
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
    Signed-off-by: Yang Bo <yangbo@linuxdeepin.com>

diff --git a/drivers/hv/hv.c b/drivers/hv/hv.c
index 6341be8..d58fac4 100644
--- a/drivers/hv/hv.c
+++ b/drivers/hv/hv.c
@@ -33,6 +33,14 @@
 #include <asm/mshyperv.h>
 #include "hyperv_vmbus.h"
 
+#ifndef PKG_ABI
+/*
+ * Preserve the ability to 'make deb-pkg' since PKG_ABI is provided
+ * by the Ubuntu build rules.
+ */
+#define PKG_ABI 0
+#endif
+
 /* The one and only */
 struct hv_context hv_context = {
 	.synic_initialized	= false,
@@ -213,7 +221,7 @@ int hv_init(void)
 	/*
 	 * Write our OS ID.
 	 */
-	hv_context.guestid = generate_guest_id(0, LINUX_VERSION_CODE, 0);
+	hv_context.guestid = generate_guest_id(0x80 /*Canonical*/, LINUX_VERSION_CODE, PKG_ABI);
 	wrmsrl(HV_X64_MSR_GUEST_OS_ID, hv_context.guestid);
 
 	/* See if the hypercall page is already set */
